### PR TITLE
fix(attest): two fixes to deletion logic

### PR DESCRIPTION
**Pull request by arajasek**
- If we find something in the cache, we should return true.
- don't delete anything if we don't have our first approved attestation

task: none

### CODE REVIEW

**Comment by arajasek** [comment]
Test to come.

**Comment by arajasek** [comment]
> Test to come.

Done.

**Review by ttarsi — APPROVED** [review]

**Review by kevinhalliday — APPROVED** [review]

**Review by corverroos — APPROVED** [review]

File: halo/attest/keeper/keeper.go [hunk 1]
```diff
@@ -796,7 +797,7 @@ func (k *Keeper) deleteBefore(ctx context.Context, height uint64) error {
 		// once we start catching up.
```
Comment (corverroos) [inline halo/attest/keeper/keeper.go:797]: maybe add a comment here, something like: `also don't delete anything before we have at least one approved attestation`

Comment (corverroos) [reply halo/attest/keeper/keeper.go:797]: Other option is to remove the boolean and just return `0` if not found, then we will not delete anything either

Comment (arajasek) [reply halo/attest/keeper/keeper.go:797]: Comment added, thanks!

**Pull request merged**